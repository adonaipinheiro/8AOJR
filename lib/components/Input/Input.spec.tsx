import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";
import { Input } from ".";
import React from "react";

describe("@components/Input", () => {
  it("should renders input with label", () => {
    render(<Input label="Username" id="username" />);

    const label = screen.getByText(/Username/i);
    expect(label).toBeInTheDocument();

    const input = screen.getByRole("textbox");
    expect(input).toBeInTheDocument();
  });

  it("should renders input with error message", () => {
    render(
      <Input
        label="Username"
        id="username"
        errorMessage="Username is required"
      />
    );

    const errorMessage = screen.getByRole("alert");
    expect(errorMessage).toHaveTextContent("Username is required");

    const input = screen.getByRole("textbox");
    expect(input).toHaveAttribute("aria-invalid", "true");
  });

  it("should has aria-describedby when there is an error", () => {
    render(
      <Input
        label="Username"
        id="username"
        errorMessage="Username is required"
      />
    );

    const input = screen.getByRole("textbox");
    const errorMessage = screen.getByRole("alert");

    expect(input).toHaveAttribute("aria-describedby", errorMessage.id);
  });

  it("should not render error message when no error is passed", () => {
    render(<Input label="Username" id="username" />);

    const errorMessage = screen.queryByRole("alert");
    expect(errorMessage).not.toBeInTheDocument();
  });

  it("should match to snapshot", () => {
    const { asFragment } = render(<Input label="Username" id="username" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
