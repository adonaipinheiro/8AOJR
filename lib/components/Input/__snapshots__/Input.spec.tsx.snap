// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`@components/Input > should match to snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gTrWKq eHFeyS"
  >
    <label
      for="username"
    >
      <p
        class="sc-dDvxFM coBwAw"
      >
        Username
      </p>
    </label>
    <input
      aria-invalid="false"
      id="username"
      name="username"
    />
  </div>
</DocumentFragment>
`;
